testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/Environment.java

```diff
@@ -103,12 +106,13 @@ private Environment(
             Map<String, DockerContainer> containers,
             EnvironmentListener listener,
             boolean attached,
-            Map<String, List<String>> configuredFeatures)
+            Map<String, List<String>> configuredFeatures,
+            List<Supplier<String>> startupLogs)
```

Comment:
why not just List<String> ?

Reply:
Because you want to evaluate this lines when containers are up to i.e. access mapped ports

Reply:
🤯